Arcade Deploy docs (#188)
ArcadeAI/docs: pages/home/configure/arcade-deploy.mdx

Author: This is kind of weird to have in the self-hosting section, but can't think of a better place for it at the moment

Reviewer: Maybe flip it around and make the section "Deploying Arcade"?

---
title: "Configuring Arcade Deploy"
description: "Learn how to configure Arcade Deploy"
---

import { Steps } from "nextra/components";
import { SignupLink } from "@/components/Analytics";

# Configuring Arcade Deploy

> Arcade Deploy is a tool that allows you to deploy your worker to the cloud.

Reviewer: We overload the term `tool` quite a bit 😛 
```suggestion
The `arcade deploy` command deploys your worker to the cloud.
```

# (Required) The unique identifier for the worker.
id = "my-example-worker"
# (Optional) Whether the worker is enabled.
enabled = true
# (Optional) The timeout for the worker for requests to the worker.
timeout = 30
# (Optional) The number of times to retry a connection to the worker before giving up.
retries = 3
# (Required) The secret for the worker.
# The default "dev" secret or empty string is not allowed.
secret = <your secret>

# (Optional) Localy packages to deploy with the worker.
# You can specify a list of directories that contain a toolkit.
# The directory must contain a pyproject.toml file or a setup.py file.
[worker.local_source]
packages = [ "./my_toolkit1", "./my_toolkit2"]

# (Optional) Pypi packages to install for the worker.
[worker.pypi_source]
packages = [ "arcade-math", "my-pypi-package"]
```

## Secrets

You can use a secret from the environment to configure the worker with the `${env: MY_SECRET}` syntax.

```bash
[[worker]]

[worker.config]
secret = "${env: MY_ENVIRONMENT_SECRET}"
```

## Using with a local engine

To register the worker with your local engine, you can use the host and port flags.

```bash
arcade deploy -h localhost -p 9099 --no-tls

> ```

Reviewer: If I want to use with Arcade Cloud, do I omit these params?

Author: yeah this is just for local
